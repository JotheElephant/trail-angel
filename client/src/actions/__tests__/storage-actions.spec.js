'use strict';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockAsyncStorage = require('mock-async-storage');

const middlewares = [thunk];

import storageActions from '../storage-actions';

const mockStore = configureStore(middlewares);

describe('storage actions', () => {
  beforeEach(() => {
    mockAsyncStorage.mock();
  });

  afterEach(() => {
    mockStore.clearActions();
    mockAsyncStorage.release();
  });

  it('loads listings from storage', () => {
    return mockStore.dispatch(storageActions.loadListingsFromStorage())
      .then((results) => {
        expect(results).toMatchSnapshot();
      });
  });

  it('saves to storage', () => {
    return mockStore.dispatch(storageActions.saveToStorage({ foo: 'bar' }))
      .then((results) => {
        expect(results).toMatchSnapshot();
      });
  });

  it('removes from storage', () => {
    return mockStore.dispatch(storageActions.removeFromStorage(123))
      .then((results) => {
        expect(results).toMatchSnapshot();
      });
  });

  it('clears all listings from storage', () => {
    return mockStore.dispatch(storageActions.clearAllListingsFromStorage())
      .then((results) => {
        expect(results).toMatchSnapshot();
      });
  });
});