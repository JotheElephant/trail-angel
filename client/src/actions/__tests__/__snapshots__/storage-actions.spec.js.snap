exports[`storage actions clears all listings from storage 1`] = `
Object {
  "type": "CLEAR_ALL_FROM_STORAGE",
}
`;

exports[`storage actions loads listings from storage 1`] = `
Object {
  "listings": Object {},
  "type": "LOAD_LISTINGS_FROM_STORAGE",
}
`;

exports[`storage actions removes from storage 1`] = `
Object {
  "id": 123,
  "type": "REMOVE_FROM_STORAGE",
}
`;

exports[`storage actions saves to storage 1`] = `
Object {
  "foo": "bar",
}
`;
